### PR TITLE
chore: add throttled limits to response

### PR DESCRIPTION
**Pull request by mkykadir**

### CODE REVIEW

File: src/lib/runtime/Web3FunctionRunner.ts [hunk 1]
```diff
@@ -267,9 +285,18 @@ export class Web3FunctionRunner {
...
+          throttled,
         };
       }
     } else {
+      if (error && error.message) {
+        throttled.memory = error.message.includes("Memory limit exceeded");
+        throttled.rpcRequest = error.message.includes(
+          "RPC requests limit exceeded"
+        );
+        throttled.duration = error.message.includes("exceed execution timeout");
```
Comment (goums) [inline src/lib/runtime/Web3FunctionRunner.ts:297]: I don't like much the `message.includes` detection.
If we update the message later, we will break this functionality without noticing it.

Maybe instead we should customize the error thrown by the runtime/sandbox when they detect the type of error.
Like adding a `error.throttledType = "memory" | "duration" | "rpcRequests"` and then testing this with a stricter typing validation

File: src/lib/runtime/Web3FunctionRunner.ts [hunk 2]
```diff
@@ -430,6 +458,15 @@ export class Web3FunctionRunner {
...
               )
             );
+          } else if (
+            (options.runtime === "docker" && signal === 137) ||
+            (options.runtime === "thread" && this._memory >= options.memory)
+          ) {
+            reject(
+              new Error(
+                `Web3Function exited with code=${signal} (Memory limit exceeded)`
+              )
```
Comment (goums) [inline src/lib/runtime/Web3FunctionRunner.ts:468]: for Example we could use an error type
`Web3FunctionRuntimeError(message: string, throttledReason?: "memory" | "duration" | "rpcRequest" )`

```suggestion
              new Web3FunctionRuntimeError(
                `Web3Function exited with code=${signal} (Memory limit exceeded)`, "memory"
              )
```

**Review by goums — CHANGES_REQUESTED** [review]

File: src/lib/runtime/Web3FunctionRunner.ts (hunk 2 above)
Comment (mkykadir) [reply src/lib/runtime/Web3FunctionRunner.ts]: fixed in a57163230e87022e98bcf7149344c8b8224518d9

File: src/lib/runtime/Web3FunctionRunner.ts [hunk 3]
```diff
@@ -267,18 +286,40 @@ export class Web3FunctionRunner {
...
+            );
+            break;
+          case "duration":
+            throttled.duration = error.message.includes(
+              "exceed execution timeout"
+            );
+            break;
+          default:
+            break;
+        }
```
Comment (goums) [inline src/lib/runtime/Web3FunctionRunner.ts:310]: We should eliminate the `message.includes` as we don't need them anymore.
What about this suggestion?
```suggestion
        throttled[error.throttledReason] = true;
```

Comment (mkykadir) [reply src/lib/runtime/Web3FunctionRunner.ts]: fixed in 7abfd4c68c102a9fea79d114886fbbe6c5017823

**Review by goums — APPROVED** [review]

**Pull request merged**